extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/SaslConfig.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.kafka.streams.runtime;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class SaslConfig {
+
+    /**
+     * JAAS login context parameters for SASL connections in the format used by JAAS configuration files
+     */
+    @ConfigItem(name = "jaas.config")
+    public Optional<String> jaasConfig;
+
+    /**
+     * The fully qualified name of a SASL client callback handler class
+     */
+    @ConfigItem(name = "client.callback.handler.class")
+    public Optional<String> clientCallbackHandlerClass;
+
+    /**
+     * The fully qualified name of a SASL login callback handler class
+     */
+    @ConfigItem(name = "login.callback.handler.class")
+    public Optional<String> loginCallbackHandlerClass;
+
+    /**
+     * The fully qualified name of a class that implements the Login interface
+     */
+    @ConfigItem(name = "login.class")
+    public Optional<String> loginClass;
+
+    /**
+     * The Kerberos principal name that Kafka runs as
+     */
+    @ConfigItem(name = "kerberos.service.name")
+    public Optional<String> kerberosServiceName;
+
+    /**
+     * Kerberos kinit command path
+     */
+    @ConfigItem(name = "kerberos.kinit.cmd")
+    public Optional<String> kerberosKinitCmd;
+
+    /**
+     * Login thread will sleep until the specified window factor of time from last refresh
+     */
+    @ConfigItem(name = "kerberos.ticket.renew.window.factor")
+    public Optional<String> kerberosTicketRenewWindowFactor;
+
+    /**
+     * Percentage of random jitter added to the renewal time
+     */
+    @ConfigItem(name = "kerberos.ticket.renew.jitter")
+    public Optional<String> kerberosTicketRenewJitter;
+
+    /**
+     * Percentage of random jitter added to the renewal time
+     */
+    @ConfigItem(name = "kerberos.min.time.before.relogin")
+    public Optional<String> kerberosMinTimeBeforeRelogin;
+
+    /**
+     * Login refresh thread will sleep until the specified window factor relative to the
+     * credential's lifetime has been reached.
+     */
+    @ConfigItem(name = "login.refresh.window.factor")
+    public Optional<String> loginRefreshWindowFactor;
+
+    /**
+     * The maximum amount of random jitter relative to the credential's lifetime
+     */
+    @ConfigItem(name = "login.refresh.window.jitter")
+    public Optional<String> loginRefreshWindowJitter;
+
+    /**
+     * The desired minimum time for the login refresh thread to wait before refreshing a credential
+     */
+    @ConfigItem(name = "login.refresh.min.period.seconds")
+    public Optional<String> loginRefreshMinPeriodSeconds;
+
+    /**
+     * The amount of buffer time before credential expiration to maintain when refreshing a credential
+     */
+    @ConfigItem(name = "login.refresh.buffer.seconds")
+    public Optional<String> loginRefreshBufferSeconds;
```

Comment:
Should be an int. But also worth double checking with Quarkus guidelines: should it be rather `Duration`, and be given as such (which would impact the option name, though, which might not be desirable as it's coming from Kafka upstream IIUC).

Reply:
Yeah all these properties should be `Duration` and we should drop the `Seconds` part. They will be documented so it's OK.